### ISSUE TITLE
[Bug]: When concurrent search, query and upsert, queryNodes oom and cp lag keeps growing

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#37156]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20241023-1d61b604-amd64
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

## server config
- qn: 2 * 8c32g
```
  config:
    dataCoord:
      compaction:
        clustering:
          autoEnable: true
      enableActiveStandby: true
    indexCoord:
      enableActiveStandby: true
    log:
      level: debug
    queryCoord:
      enableActiveStandby: true
    queryNode:
      levelZeroForwardPolicy: RemoteLoad
    rootCoord:
      enableActiveStandby: true
    trace:
      exporter: jaeger
      jaeger:
        url: http://tempo-distributor.tempo:14268/api/traces
      sampleFraction: 1
```
## test steps
1. create a collection fouram_vZxPFCS2 with 3 fields: id(varchar pk with max_length=64) + float_vector(128dim) + int64_ck (clustering_key)
2. create index for vectors
3. insert 30m entities, their pk columns are all 64 in length
4. flush -> index again -> load
5. concurrent requests: query + search + upsert
```
'concurrent_params': {'concurrent_number': [30], 'during_time': '24h', 'interval': 120}, 'concurrent_tasks': [{'type': 'search', 'weight': 15, 'params': {'nq': 100, 'top_k': 100, 'search_param': {'ef':
 120}, 'expr': None, 'guarantee_timestamp': None, 'partition_names': None, 'output_fields': ['id'], 'ignore_growing': False, 'group_by_field': None, 'timeout': 120, 'random_data': True, 'check_task': 'check_search_output', 'check_items': {'nq': 100}}}, {'type': 'query', 'wei
ght': 10, 'params': {'ids': None, 'expr': ' ', 'output_fields': None, 'offset': None, 'limit': 1000, 'ignore_growing': False, 'partition_names': None, 'timeout': 120, 'consistency_level': None, 'random_data': True, 'random_count': 500, 'random_range': [0, 50000000], 'field_name': 'int64_ck', 'field_type': 'int64', 'check_task': 'check_response', 'check_items': None}}, {'type': 'upsert', 'weight': 5, 'params': {'nb': 200, 'timeout': 60, 'random_id': False, 'random_vector': False, 'varchar_filled': True, 'start_id': 0, 'shuffle_id': False, 'check_task': 'check_response', 'check_items': None}}]}
```
## test steps
1. At 2024-10-23 15:15, non-delegator qn-b6879dc7z7vcv oom. However, from the grafana metrics we can see that the memory usage is not high 41%
![image](https://github.com/user-attachments/assets/15c5e759-6c7a-44c1-b710-f61a03fb963d)
2. Since 2024-10-23 17:12, all upsert failed with error: message send timeout: TimeoutError, and cp lag keeps growing
![image](https://github.com/user-attachments/assets/fbefd6f0-6e79-42a6-ac4d-951b72068f50)
3. At 2024-10-23 22:55，delegator qn-b6879dc7djtwr oom

[metrics of stats-varchar-cring-op-74-9005](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=P1809F7CD0C75ACF3&var-namespace=qa-milvus&var-instance=stats-varchar-cring-op-74-9005&var-collection=All&var-app_name=milvus&from=1729682647539&to=1729738327539)
 

### Steps To Reproduce

```markdown
https://argo-workflows.zilliz.cc/archived-workflows/qa/c0864a29-2250-4c3d-8c26-37dd880c1916?nodeId=zong-stats-clustering-24h-2
```

### Milvus Log

pods:
```
stats-varchar-cring-op-74-9005-etcd-0                             1/1     Running     0               19h     10.104.16.78    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-etcd-1                             1/1     Running     0               19h     10.104.26.108   4am-node32   <none>           <none>
stats-varchar-cring-op-74-9005-etcd-2                             1/1     Running     0               19h     10.104.23.18    4am-node27   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-datanode-59f9b847cd-48f65   1/1     Running     0               19h     10.104.9.37     4am-node14   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-indexnode-584f5bb9bcg6wks   1/1     Running     0               19h     10.104.6.158    4am-node13   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-indexnode-584f5bb9bcmx7mg   1/1     Running     0               19h     10.104.13.30    4am-node16   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-mixcoord-59f84774f9-48ss9   1/1     Running     0               19h     10.104.26.116   4am-node32   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-proxy-5f54749484-hfwn8      1/1     Running     0               19h     10.104.18.157   4am-node25   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-querynode-0-b6879dc7djtwr   1/1     Running     1 (8h ago)      19h     10.104.5.46     4am-node12   <none>           <none>
stats-varchar-cring-op-74-9005-milvus-querynode-0-b6879dc7z7vcv   1/1     Running     1 (16h ago)     19h     10.104.4.2      4am-node11   <none>           <none>
stats-varchar-cring-op-74-9005-minio-0                            1/1     Running     0               19h     10.104.26.109   4am-node32   <none>           <none>
stats-varchar-cring-op-74-9005-minio-1                            1/1     Running     0               19h     10.104.16.77    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-minio-2                            1/1     Running     0               19h     10.104.23.19    4am-node27   <none>           <none>
stats-varchar-cring-op-74-9005-minio-3                            1/1     Running     0               19h     10.104.34.60    4am-node37   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-bookie-0                    1/1     Running     0               19h     10.104.26.111   4am-node32   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-bookie-1                    1/1     Running     0               19h     10.104.16.79    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-bookie-2                    1/1     Running     0               19h     10.104.23.24    4am-node27   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-bookie-init-gmf9b           0/1     Completed   0               19h     10.104.16.72    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-broker-0                    1/1     Running     0               19h     10.104.16.71    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-proxy-0                     1/1     Running     0               19h     10.104.9.34     4am-node14   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-pulsar-init-rc6ht           0/1     Completed   0               19h     10.104.16.73    4am-node21   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-recovery-0                  1/1     Running     0               19h     10.104.23.16    4am-node27   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-zookeeper-0                 1/1     Running     0               19h     10.104.26.110   4am-node32   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-zookeeper-1                 1/1     Running     0               19h     10.104.23.26    4am-node27   <none>           <none>
stats-varchar-cring-op-74-9005-pulsar-zookeeper-2                 1/1     Running     0               19h     10.104.16.88    4am-node21   <none>           <none>
```
- client log path of nas: /test/fouram/log/2024_10_23/zong-stats-clustering-24h-2-1_17506

**Comment by ThreadDao** [comment]
I try to restart all pulsar pods and milvus mixcoord to prevent cp lag, but mixcoord keeps panic
![image](https://github.com/user-attachments/assets/04a1bffc-8029-4f6c-8dfb-9675f5d35f76)
[mix_8jr72.log](https://github.com/user-attachments/files/17520002/mix_8jr72.log)

**Comment by xiaocai2333** [comment]
this was been fixed, please verify.
/assign @ThreadDao 
/unassign

**Comment by ThreadDao** [comment]
fixed master-20241108-a0315783-amd64
